### PR TITLE
EdgeHub: Fix CloudProxy idle timeout

### PR DESCRIPTION
**Pull request by varunpuranik**
EdgeHub cloud proxy has a timer to close idle connections. This is because the cloud connection is not tied to the device connection anymore, so without this, the cloud connection for a device that connects and goes away will stay open forever. 

However, this does affects subscriptions like desired property updates, where we don't want to close the proxy if we have a subscription.
This PR addresses that. 

It also addresses a bug, where the timer was not stopped on closing the cloud proxy.

### CODE REVIEW

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Startup.cs [hunk 1]
```diff
@@ -127,6 +127,10 @@ IContainer BuildContainer(IServiceCollection services)
             int scopeCacheRefreshRateSecs = this.Configuration.GetValue("DeviceScopeCacheRefreshRateSecs", 3600);
             TimeSpan scopeCacheRefreshRate = TimeSpan.FromSeconds(scopeCacheRefreshRateSecs);
 
+            int cloudConnectionIdleTimeoutSecs = this.Configuration.GetValue("CloudConnectionIdleTimeoutSecs", 3600);
+            TimeSpan cloudConnectionIdleTimeout = TimeSpan.FromSeconds(cloudConnectionIdleTimeoutSecs);
+            bool closeCloudConnectionOnIdleTimeout = this.Configuration.GetValue("CloseCloudConnectionOnIdleTimeout", true);
```
Comment (damonbarry) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Startup.cs]: Why would someone change this setting to false?

File: edge-util/src/Microsoft.Azure.Devices.Edge.Util/ResettableTimer.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            var instance = new Timer(this.period.TotalMilliseconds);
+            instance.Elapsed += this.TimerOnElapsed;
+            return instance;
+        }
+
+        async void TimerOnElapsed(object sender, ElapsedEventArgs e)
+        {
+            try
+            {
+                await (this.callback?.Invoke() ?? Task.CompletedTask);
```
Comment (damonbarry) [inline edge-util/src/Microsoft.Azure.Devices.Edge.Util/ResettableTimer.cs]: In the constructor code, it looks like the callback cannot be null. So why are you handling null callback here?

Comment (varunpuranik) [reply edge-util/src/Microsoft.Azure.Devices.Edge.Util/ResettableTimer.cs]: Fair point.. will fix.

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Startup.cs (hunk 1 above)
Comment (varunpuranik) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Startup.cs]: If you don't care about having additional cloud connections.. then you might want to set it to false.

Comment (damonbarry) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Startup.cs]: Per our offline conversation, it's mostly there for debugging.

**Pull request merged**